ssl/statem/statem_lib.c

```diff
@@ -884,11 +972,13 @@ int ssl_choose_server_version(SSL *s)
      * handle version.
      */
     int server_version = s->method->version;
-    int client_version = s->client_version;
+    int client_version = hello->version;
     const version_info *vent;
     const version_info *table;
     int disabled = 0;
 
+    s->client_version = client_version;
```

Comment:
Hmmmm...the `hello` structure is currently transient and disappears once we've finished the ClientHello processing. We need `client_version` later though, so I don't think we can delete it just yet.